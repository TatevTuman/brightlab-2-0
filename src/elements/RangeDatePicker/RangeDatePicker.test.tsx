// import React from 'react'
// import { act, cleanup, render, waitFor } from '@testing-library/react'
// import userEvent, { TargetElement } from '@testing-library/user-event'
// import RangeDatePicker from './RangeDatePicker'
// import { getMonthName, getNextMonth, getPreviousMonth } from '@utils'
//
// // beforeAll(() => {})
// // afterAll(() => {})
// beforeEach(() => {
//   cleanup()
//   jest.clearAllMocks()
// })
// afterEach(() => {
//   cleanup()
//   jest.clearAllMocks()
// })
//
// describe('RangeDatePicker', () => {
//   it('renders correctly', async () => {
//     const now = new Date('January 01, 2000 00:00:00')
//     const startDate = new Date('January 01, 2000 00:00:00')
//     const endDate = new Date(now.setMonth(now.getMonth() + 1))
//
//     const props = {
//       startDate,
//       endDate
//     }
//
//     const { container } = render(<RangeDatePicker {...props} />)
//     await waitFor(() => container)
//     expect(container).toMatchSnapshot()
//   })
//
//   it('renders noToday correctly', async () => {
//     const props = {
//       noToday: true
//     }
//
//     const now = new Date()
//     const { container, getAllByText } = render(<RangeDatePicker {...props} />)
//     await waitFor(() => container)
//
//     const days = getAllByText(now.getDate().toString())
//     const today = days[0]
//
//     expect(today).toBeInTheDocument()
//     expect(today).not.toHaveClass('date-picker-day__today')
//   })
//
//   it('renders selected days correctly', async () => {
//     const now = new Date()
//     const startDate = new Date()
//     const endDate = new Date(now.setMonth(now.getMonth() + 1))
//
//     const props = {
//       startDate,
//       endDate
//     }
//
//     const { container } = render(<RangeDatePicker {...props} />)
//     await waitFor(() => container)
//
//     const selected = container.querySelectorAll('div.date-picker-day__selected')
//
//     expect(selected[0]).toBeInTheDocument()
//     expect(selected[0].innerHTML).toBe(startDate.getDate().toString())
//     expect(selected[1]).toBeInTheDocument()
//     expect(selected[0].innerHTML).toBe(endDate.getDate().toString())
//     expect(selected[2]).toBeUndefined()
//   })
//
//   it('renders disabled days correctly', async () => {
//     const props = {
//       startDate: new Date('January 01, 2000 00:00:00'),
//       endDate: new Date('February 01, 2000 00:00:00'),
//       onDatesChange: jest.fn()
//     }
//
//     const { container } = render(
//       <RangeDatePicker
//         {...props}
//         disabledDays={(day: Date) => {
//           const today = new Date()
//           // eslint-disable-next-line @typescript-eslint/ban-ts-comment
//           // @ts-ignore
//           return day - today < 0
//         }}
//       />
//     )
//     await waitFor(() => container)
//
//     const days = container.querySelectorAll('div.date-picker-day__disabled')
//
//     expect(days).toHaveLength(84)
//
//     days.forEach(day => {
//       userEvent.click(day!)
//
//       expect(props.onDatesChange).toHaveBeenCalledTimes(1)
//     })
//   })
//
//   // it('clicks on days', async () => {
//   //   const now = new Date()
//   //   const startDate = new Date()
//   //   const endDate = new Date(now.setMonth(now.getMonth() + 1))
//   //
//   //   const props = {
//   //     startDate,
//   //     endDate,
//   //     onDatesChange: jest.fn()
//   //   }
//   //
//   //   const { container } = render(<RangeDatePicker {...props} />)
//   //   await waitFor(() => container)
//   //
//   //   const selectedDays = container.querySelectorAll('div.date-picker-day__selected')
//   //   const startDay = selectedDays[0].nextSibling
//   //   const endDay = selectedDays[1].nextSibling
//   //
//   //   act(() => {
//   //     userEvent.click(startDay! as TargetElement)
//   //     userEvent.click(endDay! as TargetElement)
//   //   })
//   //
//   //   const startDayNextDay = new Date(startDate.setDate(startDate.getDate() + 1))
//   //   const endDayNextDay = new Date(endDate.setDate(endDate.getDate() + 1))
//   //
//   //   expect(props.onDatesChange).toHaveBeenCalledWith(startDayNextDay, endDate)
//   //   expect(props.onDatesChange).toHaveBeenCalledWith(startDayNextDay, endDayNextDay)
//   //
//   //   // const nextSelectedDays = container.querySelectorAll('div.date-picker-day__selected')
//   //   // const rangedDays = container.querySelectorAll('div.date-picker-day__range')
//   // })
// })
