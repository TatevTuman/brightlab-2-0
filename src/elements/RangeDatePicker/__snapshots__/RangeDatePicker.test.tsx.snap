// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeDatePicker renders correctly 1`] = `
<div>
  <div
    class="range-date-picker"
  >
    <div
      class="date-picker range-date-picker__start"
    >
      <div
        class="date-picker-header"
      >
        <div
          class="date-picker-header-title"
        >
          January
          , 
          2000
        </div>
        <nav
          class="date-picker-header-nav"
        >
          <div
            class="date-picker-header-nav__prev"
            data-disabled="false"
          >
            <test-file-stub />
          </div>
          <div
            class="date-picker-header-nav__next"
            data-disabled="false"
          >
            <test-file-stub />
          </div>
        </nav>
      </div>
      <div
        class="date-picker-days-labels"
      >
        <div
          class="date-picker-day-label"
          data-index="0"
        >
          MON
        </div>
        <div
          class="date-picker-day-label"
          data-index="1"
        >
          TUE
        </div>
        <div
          class="date-picker-day-label"
          data-index="2"
        >
          WED
        </div>
        <div
          class="date-picker-day-label"
          data-index="3"
        >
          THU
        </div>
        <div
          class="date-picker-day-label"
          data-index="4"
        >
          FRI
        </div>
        <div
          class="date-picker-day-label"
          data-index="5"
        >
          SAT
        </div>
        <div
          class="date-picker-day-label"
          data-index="6"
        >
          SUN
        </div>
      </div>
      <div
        class="date-picker-days"
      >
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (0 % 7) + 1 / span 1;"
        >
          27
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (1 % 7) + 1 / span 1;"
        >
          28
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (2 % 7) + 1 / span 1;"
        >
          29
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (3 % 7) + 1 / span 1;"
        >
          30
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (4 % 7) + 1 / span 1;"
        >
          31
        </div>
        <div
          class="date-picker-day date-picker-day__selected "
          style="grid-column: (5 % 7) + 1 / span 1;"
        >
          1
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (6 % 7) + 1 / span 1;"
        >
          2
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (7 % 7) + 1 / span 1;"
        >
          3
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (8 % 7) + 1 / span 1;"
        >
          4
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (9 % 7) + 1 / span 1;"
        >
          5
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (10 % 7) + 1 / span 1;"
        >
          6
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (11 % 7) + 1 / span 1;"
        >
          7
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (12 % 7) + 1 / span 1;"
        >
          8
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (13 % 7) + 1 / span 1;"
        >
          9
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (14 % 7) + 1 / span 1;"
        >
          10
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (15 % 7) + 1 / span 1;"
        >
          11
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (16 % 7) + 1 / span 1;"
        >
          12
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (17 % 7) + 1 / span 1;"
        >
          13
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (18 % 7) + 1 / span 1;"
        >
          14
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (19 % 7) + 1 / span 1;"
        >
          15
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (20 % 7) + 1 / span 1;"
        >
          16
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (21 % 7) + 1 / span 1;"
        >
          17
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (22 % 7) + 1 / span 1;"
        >
          18
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (23 % 7) + 1 / span 1;"
        >
          19
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (24 % 7) + 1 / span 1;"
        >
          20
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (25 % 7) + 1 / span 1;"
        >
          21
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (26 % 7) + 1 / span 1;"
        >
          22
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (27 % 7) + 1 / span 1;"
        >
          23
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (28 % 7) + 1 / span 1;"
        >
          24
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (29 % 7) + 1 / span 1;"
        >
          25
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (30 % 7) + 1 / span 1;"
        >
          26
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (31 % 7) + 1 / span 1;"
        >
          27
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (32 % 7) + 1 / span 1;"
        >
          28
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (33 % 7) + 1 / span 1;"
        >
          29
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (34 % 7) + 1 / span 1;"
        >
          30
        </div>
        <div
          class="date-picker-day date-picker-day__range "
          style="grid-column: (35 % 7) + 1 / span 1;"
        >
          31
        </div>
        <div
          class="date-picker-day date-picker-day__selected date-picker-day__disabled date-picker-day__out "
          style="grid-column: (36 % 7) + 1 / span 1;"
        >
          1
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (37 % 7) + 1 / span 1;"
        >
          2
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (38 % 7) + 1 / span 1;"
        >
          3
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (39 % 7) + 1 / span 1;"
        >
          4
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (40 % 7) + 1 / span 1;"
        >
          5
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (41 % 7) + 1 / span 1;"
        >
          6
        </div>
      </div>
    </div>
    <div
      class="date-picker range-date-picker__end"
    >
      <div
        class="date-picker-header"
      >
        <div
          class="date-picker-header-title"
        >
          February
          , 
          2000
        </div>
        <nav
          class="date-picker-header-nav"
        >
          <div
            class="date-picker-header-nav__prev"
            data-disabled="false"
          >
            <test-file-stub />
          </div>
          <div
            class="date-picker-header-nav__next"
            data-disabled="false"
          >
            <test-file-stub />
          </div>
        </nav>
      </div>
      <div
        class="date-picker-days-labels"
      >
        <div
          class="date-picker-day-label"
          data-index="0"
        >
          MON
        </div>
        <div
          class="date-picker-day-label"
          data-index="1"
        >
          TUE
        </div>
        <div
          class="date-picker-day-label"
          data-index="2"
        >
          WED
        </div>
        <div
          class="date-picker-day-label"
          data-index="3"
        >
          THU
        </div>
        <div
          class="date-picker-day-label"
          data-index="4"
        >
          FRI
        </div>
        <div
          class="date-picker-day-label"
          data-index="5"
        >
          SAT
        </div>
        <div
          class="date-picker-day-label"
          data-index="6"
        >
          SUN
        </div>
      </div>
      <div
        class="date-picker-days"
      >
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (0 % 7) + 1 / span 1;"
        >
          31
        </div>
        <div
          class="date-picker-day date-picker-day__selected "
          style="grid-column: (1 % 7) + 1 / span 1;"
        >
          1
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (2 % 7) + 1 / span 1;"
        >
          2
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (3 % 7) + 1 / span 1;"
        >
          3
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (4 % 7) + 1 / span 1;"
        >
          4
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (5 % 7) + 1 / span 1;"
        >
          5
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (6 % 7) + 1 / span 1;"
        >
          6
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (7 % 7) + 1 / span 1;"
        >
          7
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (8 % 7) + 1 / span 1;"
        >
          8
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (9 % 7) + 1 / span 1;"
        >
          9
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (10 % 7) + 1 / span 1;"
        >
          10
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (11 % 7) + 1 / span 1;"
        >
          11
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (12 % 7) + 1 / span 1;"
        >
          12
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (13 % 7) + 1 / span 1;"
        >
          13
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (14 % 7) + 1 / span 1;"
        >
          14
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (15 % 7) + 1 / span 1;"
        >
          15
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (16 % 7) + 1 / span 1;"
        >
          16
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (17 % 7) + 1 / span 1;"
        >
          17
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (18 % 7) + 1 / span 1;"
        >
          18
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (19 % 7) + 1 / span 1;"
        >
          19
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (20 % 7) + 1 / span 1;"
        >
          20
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (21 % 7) + 1 / span 1;"
        >
          21
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (22 % 7) + 1 / span 1;"
        >
          22
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (23 % 7) + 1 / span 1;"
        >
          23
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (24 % 7) + 1 / span 1;"
        >
          24
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (25 % 7) + 1 / span 1;"
        >
          25
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (26 % 7) + 1 / span 1;"
        >
          26
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (27 % 7) + 1 / span 1;"
        >
          27
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (28 % 7) + 1 / span 1;"
        >
          28
        </div>
        <div
          class="date-picker-day "
          style="grid-column: (29 % 7) + 1 / span 1;"
        >
          29
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (30 % 7) + 1 / span 1;"
        >
          1
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (31 % 7) + 1 / span 1;"
        >
          2
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (32 % 7) + 1 / span 1;"
        >
          3
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (33 % 7) + 1 / span 1;"
        >
          4
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (34 % 7) + 1 / span 1;"
        >
          5
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (35 % 7) + 1 / span 1;"
        >
          6
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (36 % 7) + 1 / span 1;"
        >
          7
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (37 % 7) + 1 / span 1;"
        >
          8
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (38 % 7) + 1 / span 1;"
        >
          9
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (39 % 7) + 1 / span 1;"
        >
          10
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (40 % 7) + 1 / span 1;"
        >
          11
        </div>
        <div
          class="date-picker-day date-picker-day__disabled date-picker-day__out "
          style="grid-column: (41 % 7) + 1 / span 1;"
        >
          12
        </div>
      </div>
    </div>
  </div>
</div>
`;
